nomos-services/api/src/lib.rs

```diff
@@ -78,17 +83,24 @@ where
 
     /// Service main loop
     async fn run(mut self) -> Result<(), DynError> {
-        let endpoint = B::new(self.settings.backend_settings).await?;
+        let mut endpoint = B::new(self.settings.backend_settings).await?;
 
         self.service_resources_handle.status_updater.notify_ready();
         tracing::info!(
             "Service '{}' is ready.",
             <RuntimeServiceId as AsServiceId<Self>>::SERVICE_ID
         );
 
+        endpoint
+            .wait_until_ready(self.service_resources_handle.overwatch_handle.clone())
+            .await?;
+
+        let endpoint = endpoint;
```

Comment:
wtf

Reply:
To be doubly clear that mutability's no longer needed. Do you reckon it's unnecessary?